xline/src/server/lock_server.rs

```diff
@@ -1,34 +1,38 @@
-use std::sync::Arc;
+use std::{marker::PhantomData, sync::Arc};
 
 use async_stream::stream;
 use clippy_utilities::OverflowArithmetic;
 use curp::{client::Client, cmd::ProposeId};
 use etcd_client::EventType;
 use tracing::debug;
 use uuid::Uuid;
+use xlineapi::RequestWithToken;
 
 use super::{
     auth_server::get_token,
-    command::{Command, CommandResponse, KeyRange, SyncResponse},
-    common::{propose, propose_indexed},
+    command::{
+        command_from_request_wrapper, propose_err_to_status, Command, CommandResponse, KeyRange,
+        SyncResponse,
+    },
 };
 use crate::{
     id_gen::IdGenerator,
     rpc::{
         Compare, CompareResult, CompareTarget, DeleteRangeRequest, DeleteRangeResponse,
         LeaseGrantRequest, LeaseGrantResponse, Lock, LockRequest, LockResponse, PutRequest,
-        RangeRequest, RangeResponse, Request, RequestOp, RequestUnion, RequestWithToken,
-        RequestWrapper, Response, ResponseHeader, SortOrder, SortTarget, TargetUnion, TxnRequest,
-        TxnResponse, UnlockRequest, UnlockResponse, WatchClient, WatchCreateRequest, WatchRequest,
+        RangeRequest, RangeResponse, Request, RequestOp, RequestUnion, RequestWrapper, Response,
+        ResponseHeader, SortOrder, SortTarget, TargetUnion, TxnRequest, TxnResponse, UnlockRequest,
+        UnlockResponse, WatchClient, WatchCreateRequest, WatchRequest,
     },
+    storage::storage_api::StorageApi,
 };
 
 /// Default session ttl
 const DEFAULT_SESSION_TTL: i64 = 60;
 
 /// Lock Server
 #[derive(Debug)]
-pub(super) struct LockServer {
+pub(super) struct LockServer<S> {
```

Comment:
could we just move the generic `S` to the `propose` method, it seems only this method need this generic.

Reply:
The main reason for adding `S` here is `command_from_request_wrapper`

Reply:
sorry I forget it's an internal method